### ISSUE TITLE
MGMT: Support attribute value filters

### ISSUE THREAD

**Issue by mmoayyed** [Unicon/cas#49]
Reference link:
https://apereo.github.io/cas/development/integration/Attribute-Value-Release-Policies.html

Presently, the only allowed option is "Regex" where the pattern is allowed by the mgmt app. Given the options are many, I wonder if this might break into its own section.

**Comment by tsschmidt** [comment]
I don't believe the management app currently handles chaining filters either.  Is this something we want to add?

**Comment by mmoayyed** [comment]
I would be OK with leaving this out, but it is not that much work...why not. Your call.

**Comment by tsschmidt** [comment]
I have pushed this to mgmt-chaining-filters branch.  Still a bit rough to merge.  Looking for feedback.

**Comment by mmoayyed** [comment]
Reviewed. Looks gorgeous to me. 

A few suggestions:

1. Probably  better to rename the titles of each filter; (with a little bit of description underneath each)
![image](https://user-images.githubusercontent.com/1205228/31486155-5d18944c-af43-11e7-94d1-699ccf0da406.png)

2. Re-arrange the cards for attribute release as such: Attribute Policy, Attribute Release Options, Attribute Release Consent, Attribute Filters, Principal Attribute Repository
3. Change "Attribute Policy" card title to "Attribute Release Policy".
4. Change "Attribute Filters" card title to "Attribute Value Filters"

**Comment by tsschmidt** [comment]
I have pushed these changes.  

A thought on Scripted Value filters.  I think it would make sense to switch the widget from either a textbox to a larger editing container depending on if they are External files or Inline groovy.  We could use a plain textarea or even get fancier with this https://ace.c9.io, that has groovy syntax highlighting.  

Implementation would mean we add a checkbox to mark at as inline or external, and then switch the editor.  This would be another flag in the ScriptedAttributeFilter.  Instead I think creating two definitions ExternalScriptedAttributeFilter and InlineScriptedAttributeFilter, would be more elegant since the user would choose when creating it and the widget wouldn't need to flip back and forth when it is visible.

**Comment by mmoayyed** [comment]
Sounds awesome.

I'll rewrite the issue description to match what you suggest; it's probably best we address these for 5.3.
